### PR TITLE
Control editor palette and gradients from theme.json

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
This PR allows theme.json to control the color palette, preset gradients.
It updates the objects in experimental-default-theme.json because meanwhile the defaults were changed.

## How has this been tested?
I verified I could change colors and gradients by changing the values on experimental-default-theme.json (provided theme, the color palette was disabled).
I verified I could set a different color palette just for the paragraph block by using theme.json.

### CODE REVIEW

File: packages/block-editor/src/store/defaults.js [hunk 1]
```diff
@@ -37,64 +36,6 @@ export const PREFERENCES_DEFAULTS = {
  */
 export const SETTINGS_DEFAULTS = {
 	alignWide: false,
-	colors: [
```
Comment (youknowriad) [inline packages/block-editor/src/store/defaults.js:40]: The removal here means someone using the block editor package without providing a color palette may not work properly, did we check on the playground? Maybe it's fine but it's an npm package breaking change we should probably mention on the changelog.

File: lib/experimental-default-theme.json [hunk 2]
```diff
@@ -4,102 +4,126 @@
...
+						"name": "Electric grass",
+						"gradient": "linear-gradient(135deg,rgb(202,248,128) 0%,rgb(113,206,126) 100%)",
+						"slug": "electric-grass"
 					},
 					{
-						"slug": "vivid-cyan-blue-to-vivid-purple",
-						"gradient": "linear-gradient(135deg,rgba(6,147,227,1) 0%,rgb(155,81,224) 100%)"
+						"name": "Midnight",
+						"gradient": "linear-gradient(135deg,rgb(2,3,129) 0%,rgb(40,116,252) 100%)",
+						"slug": "midnight"
```
Comment (youknowriad) [inline lib/experimental-default-theme.json:126]: How about i18n here? Is the idea to keep the "double registration" for now (theme.json and php)?

File: lib/edit-site-page.php [hunk 3]
```diff
@@ -132,11 +132,8 @@ function gutenberg_edit_site_init( $hook ) {
 		'maxUploadFileSize' => $max_upload_size,
 	);
 
-	list( $color_palette, ) = (array) get_theme_support( 'editor-color-palette' );
```
Comment (youknowriad) [inline lib/edit-site-page.php:135]: Do we want to offer the handling of get_theme_support and map it to the new setting like we do for the other theme support flags or is this something that is already handled.

**Review by youknowriad — COMMENTED** [review]
Glad I waited before working on this :P it was on my radar.
Great work here, I didn't test yet.

**Comment by youknowriad** [comment]
useEditorFeature definitely simplifies the client code a lot. It's a good API I think.

File: packages/block-editor/src/store/defaults.js (hunk 1 above)
Comment (jorgefilipecosta) [reply packages/block-editor/src/store/defaults.js:40]: We are going to have the WordPress defaults that the server needs to be aware of ( to generate CSS vars for example). Having defaults on the block editor inscreses the maintainability efforts. I guess we can remove the block editor defaults, I added a braking change note.

**Review by youknowriad — APPROVED** [review]
I tested this and it's working great for me.

I'd love if we explore again the automatic generated of the styles needed for the palettes on the frontend (like we do now for the CSS vars).

**Pull request merged**